models/experimental/functional_vit/tt/ttnn_optimized_sharded_vit.py

```diff
@@ -10,137 +10,106 @@
 )
 
 import ttnn
-
 from ttnn.dot_access import DotAccessDict
 
 
 def update_model_config(config, batch_size):
-    core_grid = ttnn.CoreGrid(y=8, x=12)
-
+    core_grid = ttnn.CoreGrid(y=batch_size, x=12)
+    seqL_t = int(224 / 32)  # 7
+    dim_t = int(768 / 32)  # 24
+    dim_t__x = int(dim_t / core_grid.x)  # 2
+    head_num = 12
+    head_seqL_t = int(head_num * seqL_t / core_grid.x)  # 7
+    head_size_t__x = int(dim_t / head_num)  # 2
+    class__x = int(1152 / 32 / core_grid.x)  # 3
+
+    # sharding configs
     program_configs = {
-        "fold_output_program_config": ttnn.MemoryConfig(
-            ttnn.TensorMemoryLayout.BLOCK_SHARDED,
-            ttnn.BufferType.L1,
-            ttnn.ShardSpec(
-                ttnn.CoreRangeSet(
-                    {
-                        ttnn.CoreRange(
-                            ttnn.CoreCoord(0, 0),
-                            ttnn.CoreCoord(12, 7),
-                        ),
-                    }
-                ),
-                [
-                    224,
-                    192,
-                ],
-                ttnn.ShardOrientation.ROW_MAJOR,
-                False,
-            ),
-        ),
-        "embedding_matmul_program_config": ttnn.MatmulMultiCoreReuseMultiCastProgramConfig(
-            compute_with_storage_grid_size=(core_grid.x, core_grid.y),
-            in0_block_w=3,
-            out_subblock_h=1,
-            out_subblock_w=6,
-            per_core_M=7,
-            per_core_N=6,
-            transpose_mcast=False,
-            fused_activation=None,
-        ),
         "query_key_value_matmul_program_config": ttnn.MatmulMultiCoreReuseMultiCastProgramConfig(
             compute_with_storage_grid_size=(core_grid.x, core_grid.y),
-            in0_block_w=2,
+            in0_block_w=dim_t__x,  # 2,
             out_subblock_h=1,
-            out_subblock_w=6,
-            per_core_M=7,
-            per_core_N=6,
+            out_subblock_w=int(3 * dim_t__x),  # 6,
+            per_core_M=seqL_t,  # 7,
+            per_core_N=int(3 * dim_t__x),  # 6,
```

Comment:
This is already an `int`, no need to cast.

Reply:
@yan-zaretskiy done